core/src/main/java/org/apache/iceberg/OffsetsAwareSplitScanTaskIterator.java

```diff
@@ -44,12 +39,9 @@ class OffsetsAwareSplitScanTaskIterator<T extends ScanTask> implements SplitScan
       long parentTaskLength,
       List<Long> offsetList,
       SplitScanTaskCreator<T> splitTaskCreator) {
-    Preconditions.checkArgument(
-        OFFSET_ORDERING.isStrictlyOrdered(offsetList), "Offsets must be sorted in asc order");
-
     this.parentTask = parentTask;
     this.splitTaskCreator = splitTaskCreator;
-    this.offsets = ImmutableList.copyOf(offsetList);
```

Comment:
I don't think we need to copy the list of offsets. This was introduced [here](https://github.com/apache/iceberg/pull/204#discussion_r290077463) and was flagged during the initial review.